packages/acceptance-tests/pkg-tests-core/types/global.d.ts

```diff
@@ -0,0 +1,31 @@
+interface PackageDriver {
+  (packageJson: Record<string, any>, subDefinition: Record<string, any> | RunFunction, fn?: RunFunction): any;
+  getPackageManagerName: () => string;
+  withConfig: (definition: Record<string, any>) => PackageDriver;
+}
+
+type RunFunction = (
+  {path, run, source}:
+  {
+    path: string,
+    run: (...args: any[]) => Promise<ExecResult>,
+    source: (script: string, callDefinition?: Record<string, any>) => Promise<Record<string, any>>
+  }
+) => void;
+
+type ExecResult = {
```

Comment:
Aaahh now I understand where `ExecResult` comes from ... shouldn't it be with the exec util?

Reply:
@arcanis When I tried to import anything from `global.d.ts` TS compiler stopped to recognize `global.makeTemporaryEnv`. Because of that I had to define all the types used by `makeTemporaryEnv` inside `global.d.ts`. I would love to do it other way if someone finds it.

Reply:
@arcanis I guess it would have been much easier with these types if `makeTemporaryEnv` was not defined on a global scope.

Reply:
I think there is other simpler way. We can declare `makeTemporaryEnv` in a global scope, but not on a `global` objects. This way we can place types into ts files, and get rid of `global.d.ts`. Just pushed this change

Reply:
Nah, that didn't work, I had to rollback. If I don't declare `makeTemporaryEnv` on a `global` object TS considers its type as `any`

Reply:
I think something like this might work, but it's not a blocker, we can figure it out later:

**makeTemporaryEnv.ts**

```ts
function mte(...) {
  // ...
}

global.makeTemporaryEnv = mte;

declare global {
  declare var makeTemporaryEnv: typeof mte;
}
```

Reply:
I think something like this might work, but it's not a blocker, we can figure it out later:

**makeTemporaryEnv.ts**

```ts
function mte(...) {
  // ...
}

global.makeTemporaryEnv = mte;

declare global {
  declare var makeTemporaryEnv: typeof mte;
}
```

Reply:
@arcanis Cool, that did the trick! Many thanks!